components/button/theme.css

```diff
@@ -85,261 +76,315 @@
+        color: var(--color-teal-darkest);
       }
 
       &:focus {
-        --box-shadow: inset 0 0 0 2px var(--color-teal-darkest);
+        border: 1px solid var(--color-teal-darkest);
+        box-shadow: inset 0 0 0 1px var(--color-teal-darkest);
       }
 
       &:active {
-        --background-color: color(var(--color-teal-darkest) a(5%));
-        --box-shadow: none;
+        background-color: color(var(--color-teal-darkest) a(5%));
+        box-shadow: none;
       }
     }
   }
 
   &.inverse {
-    --spinner-color: color(var(--color-neutral-lightest) a(20%));
-    --text-color: var(--color-neutral-lightest);
+    color: var(--color-neutral-lightest);
 
     &:not([disabled]) {
-      --background-color: transparent;
-
       &:hover {
-        --background-color: color(var(--color-white) a(10%));
+        background-color: color(var(--color-white) a(10%));
       }
 
       &:focus {
-        --box-shadow: inset 0 0 0 2px var(--color-neutral-lightest);
+        border: 1px solid var(--color-neutral-lightest);
+        box-shadow: inset 0 0 0 1px var(--color-neutral-lightest);
       }
 
       &:active {
-        --background-color: color(var(--color-white) a(10%));
-        --box-shadow: none;
+        background-color: color(var(--color-white) a(10%));
+        box-shadow: none;
       }
     }
   }
 }
 
 .outline {
   &:not(.inverse) {
-    --spinner-color: color(var(--color-teal-darkest) a(20%));
-    --text-color: var(--color-teal-darkest);
+    color: var(--color-teal-darkest);
 
     &[disabled] {
-      --border: 1px solid var(--color-neutral-dark);
+      border: 1px solid var(--color-neutral-dark);
     }
 
     &:not([disabled]) {
-      --background-color: transparent;
-      --border: 1px solid color(var(--color-teal-darkest) a(75%));
+      background-color: transparent;
+      border: 1px solid color(var(--color-teal-darkest) a(75%));
 
       &:hover {
-        --background-color: color(var(--color-teal-darkest) a(5%));
-        --border: 1px solid var(--color-teal-darkest);
-        --text-color: var(--color-teal-darkest);
+        background-color: color(var(--color-teal-darkest) a(5%));
+        border: 1px solid var(--color-teal-darkest);
       }
 
       &:focus {
-        --border: 1px solid var(--color-teal-darkest);
-        --box-shadow: inset 0 0 0 1px var(--color-teal-darkest);
+        border: 1px solid var(--color-teal-darkest);
+        box-shadow: inset 0 0 0 1px var(--color-teal-darkest);
       }
 
       &:active {
-        --background-color: color(var(--color-teal-darkest) a(5%));
-        --border: 1px solid var(--color-teal-darkest);
-        --box-shadow: none;
+        background-color: color(var(--color-teal-darkest) a(5%));
+        border: 1px solid var(--color-teal-darkest);
+        box-shadow: none;
       }
     }
+
+    &.processing {
+      color: transparent;
+    }
+
+    .spinner:before {
+      border: 2px solid color(var(--color-teal-darkest) a(20%));
```

Comment:
So you define 
```css
:root {
  --spinner-border-width: 2px;
}
```

but then you decide not to use it. Please be consistent; I know it's going to be a *visual* code mess when we use variables for everything, but future Dries (or someone else who will work on this) will be thankful.